src/js/Rickshaw.Graph.TapZoom.js

```diff
@@ -0,0 +1,66 @@
+Rickshaw.namespace('Rickshaw.Graph.TapZoom');
+
+Rickshaw.Graph.TapZoom = Rickshaw.Class.create({
+
+	initialize: function(args) {
+		if (!args || !args.graph) {
+			throw new Error("Rickshaw.Graph.TapZoom needs a reference to a graph");
+		}
+		var defaults = {
+			zoomAmount: 50,
+			maxZoomed: 60,
+			callback: function() {}
+		};
+
+		this.graph = args.graph;
+		this.svg = d3.select(this.graph.element).select("svg");
+		this.svgWidth = parseInt(this.svg.attr("width"), 10);
+		this.zoomAmount = args.zoomAmount || defaults.zoomAmount;
+		this.maxZoomed = args.maxZoomed || defaults.maxZoomed;
+		this.callback = args.callback || defaults.callback;
+
+		this.registerTouchEvents();
+	},
+
+	registerTouchEvents: function() {
+		var self = this;
+		var taps = [];
+		var domain = this.graph.dataDomain();
+
+		var chart = d3.select('.rickshaw_graph')
+			.style("touch-action", "manipulation");
+
+		this.svg.on("touchstart", onTouchStart);
+
+		function onTouchStart(){
+			taps.push(d3.event.timeStamp);
+
+			d3.event.preventDefault();
+
+			if (self.graph.window.xMin === undefined) {
+				self.graph.window.xMin = domain[0];
+			}
+
+			if (self.graph.window.xMax === undefined) {
+				self.graph.window.xMax = domain[1];
+			}
+
+			if (taps.length > 1 && taps[taps.length - 1] - taps[taps.length - 2] < 300) {
```

Comment:
tap.length - 2 is the same taps[0] - the array is always 2 or less, right?

Reply:
what is `300`? can you put in the variable with a good name?

Reply:
1) taps could be longer than 2 due to other incidental touches, like scrolling.
2) 300 is the margin between taps that feels good, it can certainly be put in a variable.